### ISSUE TITLE
Вынести билд-кеш стека в докере в локальную папку

### ISSUE THREAD

**Issue by antonkalinin-ml** [fullstack-development/haskell-starter-kit#28]
Есть идея монтировать папки .stack-work и ~/.stack внутри докера в локальные папки внутри рабочей копии репозитория. Что-то вроде такого:

    docker run -v .stack-work:/haskell-starter-kit/.stack-work -v .stack:/root/.stack

Эти папки образуют билд-кеш, хранимый между сборками, благодаря чему не понадобится make deps - сбилженные зависимости будут храниться локально - и make run будет быстрее засчет того, что часть модулей проекта уже сбилжена.

То, что в норме хранится в ~/.stack, можно тоже хранить как скрытую папку в рабочей копии, чтобы при ее удалении все временные файлы удалялись. Тогда у фронтендера не останется на машине никакого мусора, привнесенного стеком.

**Comment by kelizarov** [comment]
Тут надо понимать что локальный .stack-work у каждой ОС может отличаться. Насколько мне известно пытаясь запустить сбилдиные бинари макоси на линуксе могут привести к ошибке несоотвествии архиектуры платформы. Поэтому тут лучше по-экспирементирвоать.

**Comment by antonkalinin-ml** [comment]
> Тут надо понимать что локальный .stack-work у каждой ОС может отличаться. Насколько мне известно пытаясь запустить сбилдиные бинари макоси на линуксе могут привести к ошибке несоотвествии архиектуры платформы. Поэтому тут лучше по-экспирементирвоать.

Да, такое может возникнуть, если бэкендер сидит на макоси и билдит сервер двумя способами. Тогда локально он соберется под макось, а докер будет билдить под линукс. Но это можно решить разнесением на разные папки - билд-кеш докера можно положить в отдельную папку типа `.docker-build-cache/stack-work` и `.docker-build-cache/stack`.
